dashboard/src/components/Header/ContextSelector.tsx

```diff
@@ -70,7 +79,7 @@ function ContextSelector() {
     if (created) {
       closeNewNSModal();
       dispatch(actions.namespace.setNamespace(cluster, newNS));
-      dispatch(push(app.apps.list(cluster, newNS)));
+      history.push(app.apps.list(cluster, newNS));
```

Comment:
I'm confused why you're not dispatching this one, when creating a new namespace? Wouldn't we want to switch to the new namespace in that case, rather than just pushing the update to the history? Or what am I missing? What if I create a new namespace when I'm on a non-namespaced view? Or perhaps that's the intention (for a namespaced view, the state is updated due to the url causing the components to update, but for a non-namespaced it doesn't affect the current components).

Reply:
No, there is no behavioral change here. The only thing is that I am using now `useHistory` from `react-router` rather than directly `push` from `connected-react-router`. I am doing this to be able to easily mock `push` in the tests. `push` returns a `Thunkdispatch` so that's why you need to dispatch it but it's not necessary for `history.push`.

Reply:
Huh, I would have thought (but haven't checked) that `dispatch(push(...))` would be resulting in a redux `LOCATION_CHANGE` action, which we handle in certain reducers, whereas I wouldn't have thought `history.push(...)` was related to redux at all (so no action being handled.

Reply:
> whereas I wouldn't have thought `history.push(...)` was related to redux at all (so no action being handled.

`history.push` also triggers a `LOCATION_CHANGE`, in fact, any change in the URL triggers that

Reply:
OIC - so the extra `dispatch` around the `dispatch(push(...))` was not related to dispatching a `LOCATION_CHANGE` itself. Makes sense, thanks.